spinta/manifests/mermaid/helpers.py

```diff
@@ -0,0 +1,273 @@
+    multiplicity: str | None = None
+
+    ACCESS_MAPPING = {
+        Access.open: '+',
+        Access.public: '#',
+        Access.protected: '~',
+        Access.private: '-'
+    }
+
+    def __str__(self):
+
+        property_string = self.name
+
+        if self.access is not None:
+            access = self.ACCESS_MAPPING[self.access]
+            property_string = f"{access} {property_string}"
+
+        if self.type is not None:
+            property_string = f"{property_string} : {self.type}"
+
+        if self.cardinality or self.multiplicity:
+            property_string = f"{property_string} [{int(self.cardinality) if self.cardinality is not None else ''}..{self.multiplicity if self.multiplicity is not None else ''}]"
+
+        return property_string
+
+
+class RelationshipDirection(enum.Enum):
+    FORWARD = 'forward'
+    BACKWARD = 'backward'
+
+
+class RelationshipType(enum.Enum):
+    ASSOCIATION = 'association'
+    DEPENDENCY = 'dependency'
+    INHERITANCE = 'inheritance'
+
+
+@dataclass
+class MermaidRelationship:
+    node1: str
+    node2: str
+    type: RelationshipType
+    direction: RelationshipDirection = RelationshipDirection.FORWARD
+    cardinality: bool = False
+    multiplicity: str = ""
+    label: str = ""
+
+    def __str__(self):
+
+        if self.direction == RelationshipDirection.FORWARD:
+            if self.type == RelationshipType.ASSOCIATION:
+                arrow = '-->'
+            if self.type == RelationshipType.DEPENDENCY:
+                arrow = '..>'
+            if self.type == RelationshipType.INHERITANCE:
+                arrow = '--|>'
+        else:
+            if self.type == RelationshipType.ASSOCIATION:
+                arrow = '<--'
+            if self.type == RelationshipType.DEPENDENCY:
+                arrow = '<..'
+            if self.type == RelationshipType.INHERITANCE:
+                arrow = '<|--'
+
+        if self.cardinality or self.multiplicity:
+            cardinality_multiplicity = f' "[{int(self.cardinality)}..{self.multiplicity}]"'
+        else:
+            cardinality_multiplicity = ""
+
+        relationship_text = f"{self.node1} {arrow}{cardinality_multiplicity} {self.node2}"
+        if self.label:
+            relationship_text += f" : {self.label}"
+
+        return relationship_text
+
+
+def write_mermaid_manifest(context: Context, output: str, manifest: InlineManifest):
+
+    mermaids = []
+
+    for dataset_name, dataset in manifest.get_objects()["dataset"].items():
+        mermaid = MermaidClassDiagram(title=dataset_name)
+
+        models = manifest.get_objects()["model"]
+        for model in models.values():
+            model_dataset_name = model.external.dataset.name
+            if model_dataset_name == dataset_name:
+                mermaid_class = MermaidClass(name=model.basename)
+                for model_property in model.get_given_properties().values():
+
+                    # not sure if this is the best way to detect multiplicity
```

Comment:
All properties have `list` attribute:

https://github.com/atviriduomenys/spinta/blob/5f32b6c10a736db6e0ad6fd9561d5e3ed2e37775/spinta/components.py#L716

If a property is part of a list, then `list` is set to an `array` property.

Maybe this could be used, to detect multiplicity?

Reply:
čia kažkoks pasiklydęs komentaras :) Man atrodo, kad kodą, virš kurio jis buvo, ištryniau arba iškėliau kitur. Komentarą ištryniau.